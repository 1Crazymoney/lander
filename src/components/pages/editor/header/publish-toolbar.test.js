import React from 'react';
import renderer from 'react-test-renderer';

import wrapWithIntl from '../../../../utils/test-helper';

import PublishToolbar from './publish-toolbar';

it('not published', () => {
  const props = {
    user: {
      username: 'lorem',
      draft: {
        updated: '1011'
      },
      published: null
    }
  };

  const component = wrapWithIntl(<PublishToolbar {...props} />);

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('changed and not published', () => {
  const props = {
    user: {
      username: 'lorem',
      draft: {
        updated: '1011'
      },
      published: {
        updated: '10121'
      }
    }
  };

  const component = wrapWithIntl(<PublishToolbar {...props} />);

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('up to date', () => {
  const props = {
    user: {
      username: 'lorem',
      draft: {
        updated: '22222'
      },
      published: {
        updated: '22222'
      }
    }
  };

  const component = wrapWithIntl(<PublishToolbar {...props} />);

  const tree = renderer
    .create(component)
    .toJSON();

  expect(tree).toMatchSnapshot();
});