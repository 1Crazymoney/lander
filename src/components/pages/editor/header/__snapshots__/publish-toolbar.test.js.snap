// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changed and not published 1`] = `
<div
  className="publish-toolbar"
>
  <div
    className="info-content"
  >
    <div
      className="info-msg"
    >
      Last change you've made hasn't published
    </div>
    <div
      className="info-controls"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Publish
      </button>
    </div>
  </div>
</div>
`;

exports[`not published 1`] = `
<div
  className="publish-toolbar"
>
  <div
    className="info-content"
  >
    <div
      className="info-msg"
    >
      Your Lander page hasn't published yet
    </div>
    <div
      className="info-controls"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Publish
      </button>
    </div>
  </div>
</div>
`;

exports[`up to date 1`] = `
<div
  className="publish-toolbar"
>
  <a
    className="user-address"
    href="http://localhost/lorem"
    rel="noopener noreferrer"
    target="_blank"
  >
    http://localhost/lorem
  </a>
</div>
`;
